### PR TITLE
Add missing `replaceLogbackXml` in aot build configuration for gradle and maven.

### LINKED ISSUE #2418: Add missing `replaceLogbackXml` in `aot` section
In the Micronaut AOT configuration block, we miss the logback conversion:

```
aot {
   ...
   replaceLogbackXml = true
}
```

### PR DESCRIPTION
**Pull request by wetted**
closes #2418

### CODE REVIEW

**Comment by wetted** [comment]
ping @sdelamo this probably needs to change base to 4.5.x branch once it's created

**Comment by sdelamo** [comment]
@wetted Can you merge 4.4.x into this PR? see: https://github.com/micronaut-projects/micronaut-starter/pull/2419

File: starter-core/src/test/resources/expected-aot-native-image.properties [hunk 1]
```diff
@@ -11,6 +11,9 @@ precompute.environment.properties.enabled=true
 # Converts YAML configuration files to Java configuration
 yaml.to.java.config.enabled=false
 
+# Replaces logback.xml with a pure Java configuration
+logback.xml.to.java.enabled=false
```
Comment (sdelamo) [inline starter-core/src/test/resources/expected-aot-native-image.properties]:
```suggestion
logback.xml.to.java.enabled=true
```

**Review by melix — APPROVED** [review]

**Review by sdelamo — CHANGES_REQUESTED** [review]
Merge 4.4.x into this PR.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**